.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,34 @@
+name: golangci-lint
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize, reopened]
+
+# Remove all permissions from GITHUB_TOKEN except metadata.
+permissions: {}
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        working-directory:
+          - ""
+          - test
+          - api
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - name: Calculate go version
+        id: vars
+        run: echo "go_version=$(make go-version)" >> $GITHUB_OUTPUT
+      - name: Set up Go
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
+        with:
+          go-version: ${{ steps.vars.outputs.go_version }}
+      - name: golangci-lint
```

Comment:
Not sure if I understood it properly but I have followed your suggestion, lets see.

Reply:
Looks like it didn't work like I thought it would.
```
Failed to save: Unable to reserve cache with key golangci-lint.cache-2790-5ca745018e585fb21a85a94fb90af43ec2b2a651, another job may be creating this cache
```
I don't think it is important though